// @fireblocks/ts-sdk/dist/index.js
"use strict";
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FireblocksError = exports.FireblocksResponse = exports.Fireblocks = exports.BasePath = void 0;
var clientConfiguration_1 = require("./client/clientConfiguration");
Object.defineProperty(exports, "BasePath", { enumerable: true, get: function () { return clientConfiguration_1.BasePath; } });
var client_1 = require("./client/client");
Object.defineProperty(exports, "Fireblocks", { enumerable: true, get: function () { return client_1.Fireblocks; } });
var fireblocksResponse_1 = require("./response/fireblocksResponse");
Object.defineProperty(exports, "FireblocksResponse", { enumerable: true, get: function () { return fireblocksResponse_1.FireblocksResponse; } });
var fireblocksError_1 = require("./error/fireblocksError");
Object.defineProperty(exports, "FireblocksError", { enumerable: true, get: function () { return fireblocksError_1.FireblocksError; } });
__exportStar(require("./api"), exports);
__exportStar(require("./models"), exports);
