// @fireblocks/ts-sdk/dist/response/fireblocksResponse.js
"use strict";
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.convertToFireblocksResponse = exports.FireblocksResponse = void 0;
var FireblocksResponse = /** @class */ (function () {
    function FireblocksResponse(data, statusCode, headers) {
        this.data = data;
        this.statusCode = statusCode;
        this.headers = headers;
    }
    return FireblocksResponse;
}());
exports.FireblocksResponse = FireblocksResponse;
function copyHeaders(response) {
    var headersJSON = {};
    for (var _i = 0, _a = Object.entries(response.headers); _i < _a.length; _i++) {
        var _b = _a[_i], key = _b[0], value = _b[1];
        if (typeof value === 'string') {
            headersJSON[key] = value;
        }
        else if (Array.isArray(value)) {
            headersJSON[key] = value.join(',');
        }
    }
    return headersJSON;
}
function convertToFireblocksResponse(response) {
    return new FireblocksResponse(response.data, response.status, copyHeaders(response));
}
exports.convertToFireblocksResponse = convertToFireblocksResponse;
