// @fireblocks/ts-sdk/dist/models/compliance-result.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComplianceResultStatusEnum = void 0;
exports.ComplianceResultStatusEnum = {
    Started: 'Started',
    AmlStarted: 'AMLStarted',
    AmlCompleted: 'AMLCompleted',
    AmlFailed: 'AMLFailed',
    AmlinBackground: 'AMLInBackground',
    TrStarted: 'TRStarted',
    TrCompleted: 'TRCompleted',
    TrFailed: 'TRFailed',
    Completed: 'Completed',
    RegistrationStarted: 'RegistrationStarted',
    AmlRegistrationStarted: 'AMLRegistrationStarted',
    AmlRegistrationCompleted: 'AMLRegistrationCompleted',
    TrUpdateStarted: 'TRUpdateStarted',
    TrUpdateCompleted: 'TRUpdateCompleted',
    UpdateCompleted: 'UpdateCompleted',
    IncomingStarted: 'IncomingStarted',
    AmlIncomingStarted: 'AMLIncomingStarted',
    AmlIncomingCompleted: 'AMLIncomingCompleted',
    AmlIncomingFailed: 'AMLIncomingFailed',
    AmlIncomingInBackground: 'AMLIncomingInBackground',
    TrIncomingStarted: 'TRIncomingStarted',
    TrIncomingCompleted: 'TRIncomingCompleted',
    TrIncomingFailed: 'TRIncomingFailed',
    IncomingCompleted: 'IncomingCompleted'
};
