// @fireblocks/ts-sdk/dist/models/exchange-type.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExchangeType = void 0;
/**
 *
 * @export
 * @enum {string}
 */
exports.ExchangeType = {
    Binance: 'BINANCE',
    Binanceus: 'BINANCEUS',
    Bitfinex: 'BITFINEX',
    Bithumb: 'BITHUMB',
    Bitmex: 'BITMEX',
    Bitso: 'BITSO',
    Bitstamp: 'BITSTAMP',
    Bittrex: 'BITTREX',
    Circle: 'CIRCLE',
    Coinbasepro: 'COINBASEPRO',
    Coinmetro: 'COINMETRO',
    Coinspro: 'COINSPRO',
    Cryptocom: 'CRYPTOCOM',
    Deribit: 'DERIBIT',
    Ftx: 'FTX',
    Fixus: 'FIXUS',
    Gemini: 'GEMINI',
    Hitbtc: 'HITBTC',
    Huobi: 'HUOBI',
    Korbit: 'KORBIT',
    Kraken: 'KRAKEN',
    Liquid: 'LIQUID',
    Poloniex: 'POLONIEX',
    Okcoin: 'OKCOIN',
    Okex: 'OKEX',
    Seedcx: 'SEEDCX'
};
