// @fireblocks/ts-sdk/dist/models/payee-account-type.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayeeAccountType = void 0;
/**
 * - VAULT_ACCOUNT   a native Fireblocks vault account - EXCHANGE_ACCOUNT  a third-party exchange account - INTERNAL_WALLET  a whitelisted address marked as internal to the workspace/organization - EXTERNAL_WALLET a whitelisted address marked as external - NETWORK_CONNECTION a member of the Fireblocks network - FIAT_ACCOUNT a third-party account of a fiat bank (Signature, BCB, etc)
 * @export
 * @enum {string}
 */
exports.PayeeAccountType = {
    VaultAccount: 'VAULT_ACCOUNT',
    ExchangeAccount: 'EXCHANGE_ACCOUNT',
    InternalWallet: 'INTERNAL_WALLET',
    ExternalWallet: 'EXTERNAL_WALLET',
    NetworkConnection: 'NETWORK_CONNECTION',
    FiatAccount: 'FIAT_ACCOUNT'
};
