// @fireblocks/ts-sdk/dist/models/policy-rule.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyRuleDstAddressTypeEnum = exports.PolicyRuleTransactionTypeEnum = exports.PolicyRuleAmountScopeEnum = exports.PolicyRuleAmountCurrencyEnum = exports.PolicyRuleActionEnum = exports.PolicyRuleTypeEnum = void 0;
exports.PolicyRuleTypeEnum = {
    Transfer: 'TRANSFER'
};
exports.PolicyRuleActionEnum = {
    Allow: 'ALLOW',
    Block: 'BLOCK',
    _2Tier: '2-TIER'
};
exports.PolicyRuleAmountCurrencyEnum = {
    Usd: 'USD',
    Eur: 'EUR',
    Native: 'NATIVE'
};
exports.PolicyRuleAmountScopeEnum = {
    SingleTx: 'SINGLE_TX',
    Timeframe: 'TIMEFRAME'
};
exports.PolicyRuleTransactionTypeEnum = {
    Transfer: 'TRANSFER',
    ContractCall: 'CONTRACT_CALL',
    Approve: 'APPROVE',
    Mint: 'MINT',
    Burn: 'BURN',
    Supply: 'SUPPLY',
    Redeem: 'REDEEM',
    Stake: 'STAKE',
    Raw: 'RAW',
    TypedMessage: 'TYPED_MESSAGE'
};
exports.PolicyRuleDstAddressTypeEnum = {
    Whitelisted: 'WHITELISTED',
    OneTime: 'ONE_TIME',
    Star: '*'
};
