// @fireblocks/ts-sdk/dist/models/policy-src-or-dest-sub-type.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicySrcOrDestSubType = void 0;
/**
 * * EXTERNAL - A whitelisted wallet assigned as external is typically used for addresses managed by your clients and counterparties * INTERNAL - A whitelisted wallet assigned as internal, is typically used for addresses that you control outside of your Fireblocks workspace * CONTRACT - A whitelisted wallet assigned as contract is for identifying and managing external smart contracts * EXCHANGETEST - Exchanges which operate only on testnet assets * \"*\" - All subtypes
 * @export
 * @enum {string}
 */
exports.PolicySrcOrDestSubType = {
    External: 'EXTERNAL',
    Internal: 'INTERNAL',
    Contract: 'CONTRACT',
    Exchangetest: 'EXCHANGETEST',
    Star: '*'
};
