// @fireblocks/ts-sdk/dist/models/screening-provider-rules-configuration-response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScreeningProviderRulesConfigurationResponseStatusEnum = exports.ScreeningProviderRulesConfigurationResponseDirectionEnum = exports.ScreeningProviderRulesConfigurationResponseActionEnum = void 0;
exports.ScreeningProviderRulesConfigurationResponseActionEnum = {
    Accept: 'ACCEPT',
    Reject: 'REJECT',
    Alert: 'ALERT',
    Wait: 'WAIT',
    Freeze: 'FREEZE',
    Cancel: 'CANCEL'
};
exports.ScreeningProviderRulesConfigurationResponseDirectionEnum = {
    Inbound: 'INBOUND',
    Outbound: 'OUTBOUND'
};
exports.ScreeningProviderRulesConfigurationResponseStatusEnum = {
    Completed: 'COMPLETED',
    Pending: 'PENDING',
    Rejected: 'REJECTED',
    Failed: 'FAILED',
    Canceled: 'CANCELED',
    BlockingTimeExpired: 'BLOCKING_TIME_EXPIRED'
};
