// @fireblocks/ts-sdk/dist/models/smart-transfer-ticket-term.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmartTransferTicketTermStatusEnum = exports.SmartTransferTicketTermTxStatusEnum = void 0;
exports.SmartTransferTicketTermTxStatusEnum = {
    Queued: 'queued',
    Submitted: 'submitted',
    PendingSignature: 'pendingSignature',
    PendingAuthorization: 'pendingAuthorization',
    Broadcasting: 'broadcasting',
    Confirming: 'confirming',
    PendingAmlScreening: 'pendingAmlScreening',
    Completed: 'completed',
    Cancelled: 'cancelled',
    Failed: 'failed',
    Rejected: 'rejected',
    Blocked: 'blocked',
    ThirdPartyPendingManualApproval: 'thirdPartyPendingManualApproval',
    ThirdPartyPending: 'thirdPartyPending',
    PartiallyCompleted: 'partiallyCompleted',
    Cancelling: 'cancelling'
};
exports.SmartTransferTicketTermStatusEnum = {
    Created: 'CREATED',
    Funding: 'FUNDING',
    FundingFailed: 'FUNDING_FAILED',
    Funded: 'FUNDED',
    Rejected: 'REJECTED'
};
