// @fireblocks/ts-sdk/dist/models/smart-transfer-ticket.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmartTransferTicketDvpExecutionStatusEnum = exports.SmartTransferTicketDirectionEnum = exports.SmartTransferTicketStatusEnum = exports.SmartTransferTicketTypeEnum = void 0;
exports.SmartTransferTicketTypeEnum = {
    Async: 'ASYNC',
    Dvp: 'DVP'
};
exports.SmartTransferTicketStatusEnum = {
    Draft: 'DRAFT',
    PendingApproval: 'PENDING_APPROVAL',
    Open: 'OPEN',
    InSettlement: 'IN_SETTLEMENT',
    Fulfilled: 'FULFILLED',
    Expired: 'EXPIRED',
    Canceled: 'CANCELED'
};
exports.SmartTransferTicketDirectionEnum = {
    Exchange: 'EXCHANGE',
    Send: 'SEND',
    Receive: 'RECEIVE',
    Intermediate: 'INTERMEDIATE'
};
exports.SmartTransferTicketDvpExecutionStatusEnum = {
    Started: 'STARTED',
    CreatingOrder: 'CREATING_ORDER',
    OrderCreated: 'ORDER_CREATED',
    Fulfilling: 'FULFILLING',
    FulfillingOrderFailed: 'FULFILLING_ORDER_FAILED',
    CreatingOrderFailed: 'CREATING_ORDER_FAILED',
    Fulfilled: 'FULFILLED'
};
