// @fireblocks/ts-sdk/dist/models/token-response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenResponseBlockchainDescriptorEnum = void 0;
exports.TokenResponseBlockchainDescriptorEnum = {
    Eth: 'ETH',
    EthTest3: 'ETH_TEST3',
    EthTest5: 'ETH_TEST5',
    EthTest6: 'ETH_TEST6',
    Polygon: 'POLYGON',
    PolygonTestMumbai: 'POLYGON_TEST_MUMBAI',
    AmoyPolygonTest: 'AMOY_POLYGON_TEST',
    Xtz: 'XTZ',
    XtzTest: 'XTZ_TEST',
    BasechainEth: 'BASECHAIN_ETH',
    BasechainEthTest3: 'BASECHAIN_ETH_TEST3',
    Etherlink: 'ETHERLINK',
    EtherlinkTest: 'ETHERLINK_TEST',
    Mantle: 'MANTLE',
    MantleTest: 'MANTLE_TEST',
    GunGunzillaTest: 'GUN_GUNZILLA_TEST'
};
