// @fireblocks/ts-sdk/dist/models/trading-account-type.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TradingAccountType = void 0;
/**
 *
 * @export
 * @enum {string}
 */
exports.TradingAccountType = {
    CoinFutures: 'COIN_FUTURES',
    CoinMarginedSwap: 'COIN_MARGINED_SWAP',
    Exchange: 'EXCHANGE',
    Funding: 'FUNDING',
    Fundable: 'FUNDABLE',
    Futures: 'FUTURES',
    FuturesCross: 'FUTURES_CROSS',
    Margin: 'MARGIN',
    MarginCross: 'MARGIN_CROSS',
    Options: 'OPTIONS',
    Spot: 'SPOT',
    UsdtMarginedSwapCross: 'USDT_MARGINED_SWAP_CROSS',
    UsdtFutures: 'USDT_FUTURES',
    Unified: 'UNIFIED'
};
